src/GitTabExpansion.ps1

```diff
@@ -69,6 +69,22 @@ $script:gitCommandsWithParamValues = $gitParamValues.Keys -join '|'
 $script:vstsCommandsWithShortParams = $shortVstsParams.Keys -join '|'
 $script:vstsCommandsWithLongParams = $longVstsParams.Keys -join '|'
 
+# The regular expression here is roughly follows this pattern:
+#
+# <begin anchor><whitespace>*<git>(<whitespace><parameter>)*<whitespace>+<$args><whitespace>*<end anchor>
+#
+# The delimiters inside the parameter list and between some of the elements are non-newline whitespace characters ([^\S\r\n]).
+# In those instances, newlines are only allowed if they preceded by a non-newline whitespace character.
+#
+# Begin anchor (^|[;`n])
+# Whitespace   (\s*)
+# Git Command  (?<cmd>$(GetAliasPattern git))
+# Parameters   (?<params>(([^\S\r\n]|[^\S\r\n]``\r?\n)+\S+)*)
+# $args Anchor (([^\S\r\n]|[^\S\r\n]``\r?\n)+\`$args)
+# Whitespace   (\s|``\r?\n)*
+# End Anchor   ($|[|;`n])
+$script:GitProxyFunctionRegex = "(^|[;`n])(\s*)(?<cmd>$(Get-AliasPattern git))(?<params>(([^\S\r\n]|[^\S\r\n]``\r?\n)+\S+)*)(([^\S\r\n]|[^\S\r\n]``\r?\n)+\`$args)(\s|``\r?\n)*($|[|;`n])"
```

Comment:
I wonder if we should define this regex in the GitTabSettings at the tops of this script:
```
$Global:GitTabSettings = New-Object PSObject -Property @{
    AllCommands = $false
    KnownAliases = @{
        '!f() { exec vsts code pr "$@"; }; f' = 'vsts.pr'
    }
    EnableLogging = $false
    LogPath = Join-Path ([System.IO.Path]::GetTempPath()) posh-git_tabexp.log
}
```
If that regex didn't work for a particular user's proxy function, they could then tweak the regex without requiring a turn of posh-git.

Reply:
This sounds like a good idea, and it should work for the legacy and PowerTab implementations.  However, I don't think it would help for the `Register-ArgumentCompleter` implementation.  We would need to set the regular expression at module load time for the setting to have any effect.  Maybe there should be some sort of refactoring to allow settings to be set before the module is loaded.  Maybe allow the user to set a hash table that is marshalled into settings type at module load?